domain/machine/state/machine_cloud_instance.go

```diff
@@ -32,18 +32,29 @@ func (st *State) HardwareCharacteristics(
 		return nil, errors.Trace(err)
 	}
 	retrieveHardwareCharacteristics := `
-SELECT &instanceData.*
-FROM   machine_cloud_instance
-WHERE  machine_uuid = $instanceData.machine_uuid`
-	machineUUIDQuery := instanceData{
+SELECT    (machine_cloud_instance.machine_uuid,
+           machine_cloud_instance.instance_id,
+           machine_cloud_instance.arch,
+           machine_cloud_instance.mem,
+           machine_cloud_instance.root_disk,
+           machine_cloud_instance.root_disk_source,
+           machine_cloud_instance.cpu_cores,
+           machine_cloud_instance.cpu_power,
+           machine_cloud_instance.virt_type) AS (&instanceDataResult.*),
+           availability_zone.name AS &instanceDataResult.availability_zone
```

Comment:
Would `SELECT machine_cloud_instance.* AS &instanceDataResult` work?

Reply:
No, because we have `availability_zone.name` coming from another table, so we need to use `AS`, and that breaks the whole struct.

Reply:
Make a view. Then do what @jack-w-shaw said.

Reply:
Sure, nice idea. Done in d468ec13d0.